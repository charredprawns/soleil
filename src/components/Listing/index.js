import React from 'react'
import styles from './styles'
import { Paper, Typography, withStyles } from '@material-ui/core'
import PropTypes from 'prop-types'
import clsx from 'clsx'
import { addEllipses } from 'utils'

import PeopleIcon from '@material-ui/icons/People'
import HotelIcon from '@material-ui/icons/Hotel'
import MeetingRoomIcon from '@material-ui/icons/MeetingRoom'
import BathtubIcon from '@material-ui/icons/Bathtub'

/**
@param {classes} Object: Used to define style classes. DOCUMENT YOUR CODE! :)
@param {listingData} Object: Data about a listing
@param {className} String: Wrapping class for component
*/

const IconWithNumber = ({ classes, icon, data }) => {
  return (
    <Typography className={clsx(classes.text, classes.grid_item)}>
      {data} {icon}
    </Typography>
  )
}

const Listing = ({ classes, listingData, className }) => {
  const placeholder = 'https://source.unsplash.com/random/300x150'
  return (
    <Paper elevation={5} className={clsx(classes.container, className)}>
      <img alt='random from unsplash' src={placeholder} />
      <Typography
        paragraph
        align='center'
        variant='h5'
        className={classes.title}
      >
        {listingData[1].name}
      </Typography>
      <div className={classes.grid_info}>
        <IconWithNumber
          classes={classes}
          icon={<PeopleIcon className={classes.icon} />}
          data={listingData[1].accommodates}
        />
        <IconWithNumber
          classes={classes}
          icon={<HotelIcon className={classes.icon} />}
          data={listingData[1].beds}
        />
        <IconWithNumber
          classes={classes}
          icon={<MeetingRoomIcon className={classes.icon} />}
          data={listingData[1].bedrooms}
        />
        <IconWithNumber
          classes={classes}
          icon={<BathtubIcon className={classes.icon} />}
          data={listingData[1].bathrooms}
        />
      </div>
      <Typography paragraph className={classes.text}>
        {addEllipses(listingData[1].summary, 105)}
      </Typography>

      <a
        href={listingData[1].listing_url}
        target='_blank'
        rel='noopener noreferrer'
        className={classes.link}
      >
        <Typography align='right'>See Actual Listing</Typography>
      </a>
    </Paper>
  )
}

Listing.propTypes = {
  classes: PropTypes.object
}

export default withStyles(styles)(Listing)
