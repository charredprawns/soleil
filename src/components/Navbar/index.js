import React from 'react'
import { useTheme, useMediaQuery } from '@material-ui/core'
import DesktopNavbar from './DesktopNavbar'
import MobileNavbar from './MobileNavbar'

const ManagerTemplate = ({ children, ...props }) => {
  const theme = useTheme()
  const isDesktop = useMediaQuery(theme.breakpoints.up('md'))

  return (
    <>
      {isDesktop ? (
        <>
          <DesktopNavbar {...props} />
          {children}
        </>
      ) : (
        <>
          {children}
          <MobileNavbar {...props} />
        </>
      )}
    </>
  )
}

export default ManagerTemplate
