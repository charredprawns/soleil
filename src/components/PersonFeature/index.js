import React from 'react'
import styles from './styles'
import { withStyles, Typography } from '@material-ui/core'
import PropTypes from 'prop-types'

import mainImage from 'assets/goat.jpg'
/**
 *
 * @param {Boolean} shouldSpin Should the spinning animation be active?
 */
const PersonFeature = ({ classes, shouldSpin }) => {
  return (
    <div className={classes.container}>
      <img src={mainImage} className={classes.main_image} alt='MAR' />
      <Typography variant='h1'>
        <b>MAR</b>
      </Typography>
    </div>
  )
}

PersonFeature.propTypes = {
  classes: PropTypes.object
}

PersonFeature.defaultProps = {
  shouldSpin: false
}

export default withStyles(styles)(PersonFeature)
