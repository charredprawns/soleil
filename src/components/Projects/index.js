import React from 'react'
import styles from './styles'
import PropTypes from 'prop-types'
import { withStyles, Typography } from '@material-ui/core'
import { Link } from 'react-router-dom'

/**
@param {classes} Object: Used to define style classes. DOCUMENT YOUR CODE! :)
@param {projects} Object: An array of objects, each of which constitures a project
*/
const Projects = ({ classes, projects }) => {
  const placeholder = 'https://via.placeholder.com/300x150.png?text='
  return (
    <div className={classes.container}>
      <div className={classes.grid}>
        {projects.map((project, i) => {
          return (
            <div key={i} className={classes.project}>
              <img
                src={`${placeholder}+${project.name}`}
                alt='placeholder'
                className={classes.projectImage}
              />
              <div className={classes.projectText}>
                <Typography variant='h5'>{project.name}</Typography>
                <Typography variant='body1'>
                  <a
                    href={project.url}
                    target='_blank'
                    rel='noopener noreferrer'
                  >
                    {project.url}
                  </a>
                </Typography>
              </div>
            </div>
          )
        })}
      </div>
    </div>
  )
}

Projects.propTypes = {
  classes: PropTypes.object
}

export default withStyles(styles)(Projects)
