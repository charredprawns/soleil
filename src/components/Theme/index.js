import React, { useState, useEffect } from 'react'
import { MuiThemeProvider, useMediaQuery } from '@material-ui/core'
import { ThemeProvider } from '@material-ui/styles'
import PropTypes from 'prop-types'
import makeTheme from './makeTheme'

// TODO: Changing this variable should update the theme. Move into redux
const theme = 'conformity'

const Theme = ({ children }) => {
  const isMobile = useMediaQuery('@media only screen and (max-width: 709px)')
  const isTablet = useMediaQuery(
    '@media only screen and (max-width: 969px) and (min-width: 709px)'
  )
  const isDesktop = useMediaQuery('@media only screen and (min-width: 969px)')
  const [muiTheme, setMuiTheme] = useState(
    makeTheme({
      themeName: theme,
      screenSize: isDesktop ? 'desktop' : isTablet ? 'tablet' : 'mobile'
    })
  )

  // Recreate the theme with new colors when screen size changes
  useEffect(() => {
    setMuiTheme(
      makeTheme({
        themeName: theme,
        screenSize: isDesktop ? 'desktop' : isTablet ? 'tablet' : 'mobile'
      })
    )
  }, [isMobile, isTablet, isDesktop])

  return (
    muiTheme && (
      <MuiThemeProvider theme={muiTheme}>
        <ThemeProvider theme={muiTheme}>{children}</ThemeProvider>
      </MuiThemeProvider>
    )
  )
}

Theme.propTypes = {
  children: PropTypes.any
}

export default Theme
