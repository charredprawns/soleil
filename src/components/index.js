export { default as Theme } from './Theme'
export { default as Navbar } from './Navbar'
export { default as PersonFeature } from './PersonFeature'
export { default as Projects } from './Projects'
export { default as Listing } from './Listing'
