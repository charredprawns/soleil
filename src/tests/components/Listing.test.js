import React from 'react'
import { Listing } from 'components'
import renderer from 'react-test-renderer'
describe('Listing component', () => {
  it('renders correctly', () => {
    const listingData = [
      'rand',
      {
        name: 'Big title',
        accomodates: '2',
        beds: '3',
        bedrooms: '11',
        bathrooms: '4',
        summary: 'This is a great place to live',
        listing_url: 'http://agreatlink.com'
      }
    ]
    const ListingComponent = renderer
      .create(<Listing listingData={listingData} />)
      .toJSON()

    expect(ListingComponent).toMatchSnapshot()
  })
})
