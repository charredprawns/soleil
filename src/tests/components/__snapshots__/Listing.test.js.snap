// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listing component render correctly 1`] = `
<div
  className="MuiPaper-root Listing-container-1 MuiPaper-elevation5 MuiPaper-rounded"
>
  <img
    alt="random from unsplash"
    src="https://source.unsplash.com/random/300x150"
  />
  <p
    className="MuiTypography-root Listing-title-6 MuiTypography-h5 MuiTypography-paragraph MuiTypography-alignCenter"
  >
    Big title
  </p>
  <div
    className="Listing-grid_info-2"
  >
    <p
      className="MuiTypography-root Listing-grid_item-3 MuiTypography-body1"
    >
       
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root Listing-icon-4"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
        />
      </svg>
    </p>
    <p
      className="MuiTypography-root Listing-grid_item-3 MuiTypography-body1"
    >
      3
       
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root Listing-icon-4"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 13c1.66 0 3-1.34 3-3S8.66 7 7 7s-3 1.34-3 3 1.34 3 3 3zm12-6h-8v7H3V5H1v15h2v-3h18v3h2v-9c0-2.21-1.79-4-4-4z"
        />
      </svg>
    </p>
    <p
      className="MuiTypography-root Listing-grid_item-3 MuiTypography-body1"
    >
      11
       
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root Listing-icon-4"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M14 6v15H3v-2h2V3h9v1h5v15h2v2h-4V6h-3zm-4 5v2h2v-2h-2z"
        />
      </svg>
    </p>
    <p
      className="MuiTypography-root Listing-grid_item-3 MuiTypography-body1"
    >
      4
       
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root Listing-icon-4"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <circle
          cx="7"
          cy="7"
          r="2"
        />
        <path
          d="M20 13V4.83C20 3.27 18.73 2 17.17 2c-.75 0-1.47.3-2 .83l-1.25 1.25c-.16-.05-.33-.08-.51-.08-.4 0-.77.12-1.08.32l2.76 2.76c.2-.31.32-.68.32-1.08 0-.18-.03-.34-.07-.51l1.25-1.25c.15-.15.36-.24.58-.24.46 0 .83.37.83.83V13h-6.85c-.3-.21-.57-.45-.82-.72l-1.4-1.55c-.19-.21-.43-.38-.69-.5-.31-.15-.65-.23-1-.23C6 10.01 5 11.01 5 12.25V13H2v6c0 1.1.9 2 2 2 0 .55.45 1 1 1h14c.55 0 1-.45 1-1 1.1 0 2-.9 2-2v-6h-2z"
        />
      </svg>
    </p>
  </div>
  <p
    className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
  >
    This is a great place to live
  </p>
  <a
    className="Listing-link-5"
    href="http://agreatlink.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight"
    >
      See Actual Listing
    </p>
  </a>
</div>
`;

exports[`Listing component renders correctly 1`] = `
<div
  className="MuiPaper-root Listing-container-1 MuiPaper-elevation5 MuiPaper-rounded"
>
  <img
    alt="random from unsplash"
    src="https://source.unsplash.com/random/300x150"
  />
  <p
    className="MuiTypography-root Listing-title-6 MuiTypography-h5 MuiTypography-paragraph MuiTypography-alignCenter"
  >
    Big title
  </p>
  <div
    className="Listing-grid_info-2"
  >
    <p
      className="MuiTypography-root Listing-grid_item-3 MuiTypography-body1"
    >
       
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root Listing-icon-4"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M16 11c1.66 0 2.99-1.34 2.99-3S17.66 5 16 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm-8 0c1.66 0 2.99-1.34 2.99-3S9.66 5 8 5C6.34 5 5 6.34 5 8s1.34 3 3 3zm0 2c-2.33 0-7 1.17-7 3.5V19h14v-2.5c0-2.33-4.67-3.5-7-3.5zm8 0c-.29 0-.62.02-.97.05 1.16.84 1.97 1.97 1.97 3.45V19h6v-2.5c0-2.33-4.67-3.5-7-3.5z"
        />
      </svg>
    </p>
    <p
      className="MuiTypography-root Listing-grid_item-3 MuiTypography-body1"
    >
      3
       
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root Listing-icon-4"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 13c1.66 0 3-1.34 3-3S8.66 7 7 7s-3 1.34-3 3 1.34 3 3 3zm12-6h-8v7H3V5H1v15h2v-3h18v3h2v-9c0-2.21-1.79-4-4-4z"
        />
      </svg>
    </p>
    <p
      className="MuiTypography-root Listing-grid_item-3 MuiTypography-body1"
    >
      11
       
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root Listing-icon-4"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M14 6v15H3v-2h2V3h9v1h5v15h2v2h-4V6h-3zm-4 5v2h2v-2h-2z"
        />
      </svg>
    </p>
    <p
      className="MuiTypography-root Listing-grid_item-3 MuiTypography-body1"
    >
      4
       
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root Listing-icon-4"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <circle
          cx="7"
          cy="7"
          r="2"
        />
        <path
          d="M20 13V4.83C20 3.27 18.73 2 17.17 2c-.75 0-1.47.3-2 .83l-1.25 1.25c-.16-.05-.33-.08-.51-.08-.4 0-.77.12-1.08.32l2.76 2.76c.2-.31.32-.68.32-1.08 0-.18-.03-.34-.07-.51l1.25-1.25c.15-.15.36-.24.58-.24.46 0 .83.37.83.83V13h-6.85c-.3-.21-.57-.45-.82-.72l-1.4-1.55c-.19-.21-.43-.38-.69-.5-.31-.15-.65-.23-1-.23C6 10.01 5 11.01 5 12.25V13H2v6c0 1.1.9 2 2 2 0 .55.45 1 1 1h14c.55 0 1-.45 1-1 1.1 0 2-.9 2-2v-6h-2z"
        />
      </svg>
    </p>
  </div>
  <p
    className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
  >
    This is a great place to live
  </p>
  <a
    className="Listing-link-5"
    href="http://agreatlink.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight"
    >
      See Actual Listing
    </p>
  </a>
</div>
`;
